File: 2239-find-num-closest-to-zero/findnum.ts
function findClosestNumber(nums: number[]): number {
    // start with num higher than anything
    let closest = Infinity;
  
    for (let num of nums) {
        // if num is lower than closest, update closest to that num, then move on 
      if (Math.abs(num) < Math.abs(closest)) {
        closest = num
      } else if (Math.abs(num) === Math.abs(closest)) {
          // if num is closest, return
        return Math.abs(closest)
      }
    }
  
    return closest;
  };

File: 2239-find-num-closest-to-zero/findnum.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        m = 10 ** 6
        for i in nums:
            x = abs(i-0)
            if x < m:
                m = x
                val = i
            elif x == m and val < i:
                val = i
        return val